### PR TITLE
Code Cleaning and Invalid Audiofiles Handling

### PR DESCRIPTION
**Pull request by sezanzeb**
**1. gtk tool, add folder and broken/non audiofiles**

When I put a text file into the sound source directory and renamed it to foobar.flac, no other flac files from other subdirectories were added if this textfile was the first .flac file seen. This was caused, because the TypeFinder only looked at one single sample of .flac files to detect the mime type. Hence the .flac type was not added to the list of accepted file extensions.

This change checks the mime type of all the sound files instead in order to find out which ones are broken. After this fix, all the .flac files are added to the list for conversion except those that don't contain sound (those for which gstreamer throws errors)

This increases the time to read in large folders, but it is the way for integrity, which is the first priority when handling files. Also, reading in the files is still fast compared to the conversion. Because of this longer required time for reading in files, the statusbar now also shows the progress of reading in the files.

An Error message appears to show how many files (only those with audio extensions) failed. The ErrorDialog apperance was also fixed.

old:
old_dialog

new:
new_dialog

After reading this message, the user can decide whether or not to start the conversion.

**2. Cleanup Changelog**

- added some docstrings
- removed a dead path from .gitignore
- removed 7 year old TODO file in a foreign language. If those TODOs are still relevant, can you please create issues in github in english?
- added building and installation information to readme
- used vulture and removed some unused variables and imports (after checking if it really is unused)
- added missing whitespaces after commas in lists and functions
- formatted docstring whitespaces into a uniform format across all python files
- removed all trailing whitespaces in python files

**3. soundconverter.py**

- moved imports and such things to the top, dividing it from the mode, command line arg and settings handling. It is considered better practice to have them on the top.
- if `from soundconverter.ui import gui_main` fails, it would switch to batch mode. However, this would start conversion immediatelly, either with no command line arguments at all or some files, which would convert all the files into their folders without further user configuration. This is sudden unexpected behaviour and therefore I would rather go a different way. The import of gui_main will fail if there is an e.g. syntax error in that file. So the proper way would be not make sure gui_main doesn't throw errors upon importing. I moved this import to the top.
- grouped batch options in --help using an OptionParser group. Also some modifications to the help output for batch parameters.

### CODE REVIEW

**Comment by sezanzeb** [comment]
I think the best way would be to store a list of broken files in some sort of logfile, but where would you put it?

**Comment by sezanzeb** [comment]
Gonna add a menu entry to the dropdown to open the list of the broken files later on

**Comment by sezanzeb** [comment]
> Gonna add a menu entry to the dropdown to open the list of the broken files later on, for the user to see more easily what was broken

added the described features:

couldnotread
invalidfiles
listinvalid

clear list also clears the lits of invalid files. The displayed list is a TextView, so that the list can be easily copied and saved elsewhere if desired.

File: soundconverter/ui.py [hunk 1]
```diff
@@ -276,20 +275,49 @@ def add_uris(self, uris, base=None, extensions=None):
...
             if sound_file.uri in self.filelist:
                 log('file already present: \'%s\'' % sound_file.uri)
                 continue
             self.append_file(sound_file)
 
+        if broken_audiofiles > 0:
+            s = ''
+            if broken_audiofiles > 1:
+                s = 's'
+            show_error('{} audiofile{} could not be read by gstreamer!'.format(broken_audiofiles, s),
```
Comment (kassoulet) [inline soundconverter/ui.py]: This will be very problematic to translate.

Comment (kassoulet) [reply soundconverter/ui.py]: You have to use ngettext() for this.

Comment (sezanzeb) [reply soundconverter/ui.py]: fixed in fd173843bb14eb9ba7e92c7a0300e992723db731

**Comment by kassoulet** [comment]
Well done. Thank you!

**Pull request merged**